containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -314,33 +338,46 @@ def metadata_path(self):
         )
         return BucketPath(bucket=self.metadata_bucket, key=key)
 
+    def schema_path(self, table_name: str) -> BucketPath:
+        """
+        Path to the latest version schema file for a given table
+        """
+        key = os.path.join(
+            self.name,
+            self.latest_version,
+            table_name,
+            "schema.json",
+        )
+        return BucketPath(bucket=self.metadata_bucket, key=key)
+
     @staticmethod
-    def metadata_spec_prefix(bucket_name: str | None = None) -> BucketPath:
+    def specification_prefix(metadata_type: str, bucket_name: str | None = None) -> BucketPath:
         """
-        Path to the metadata spec files
+        Path to prefix of the schema or metadata specification json schema, before version
         """
+        spec_type = MetadataTypes(metadata_type)
         return BucketPath(
             bucket_name or get_metadata_bucket(),
             os.path.join(
-                "data_product_metadata_spec",
+                f"{spec_type.name}_spec",
             ),
         )
 
     @staticmethod
-    def metadata_spec_path(version: str, bucket_name: str | None = None) -> BucketPath:
+    def specification_path(metadata_type: str, version: str, bucket_name: str | None = None) -> BucketPath:
```

Comment:
I like the idea of this handling both the JSON schema for the glue schemas and the JSON schema for the metadata. 👍🏻 

Maybe rather than making this a static method we could move it to a top level function though, I feel like it's quite separate from what the dataclass is doing.